src/components/Header/style.tsx

```diff
@@ -118,7 +118,7 @@ export type ButtonProps = {
   LayoutProps &
   JustifyContentProps
 
-export const Button = styled.div<ButtonProps>(
+export const Button = styled.button<ButtonProps>(
```

Comment:
Button in this context is better for accessibility and I have tested to ensure it looks and behaves the same. 
Let me know if there is a reason I missed for this not being a HTML button